ocs_ci/ocs/longevity.py

```diff
@@ -629,6 +666,120 @@ def stage_2(
             )
             time.sleep(delay)
 
+    def stage3(
+        self,
+        project_factory,
+        num_of_pvc,
+        num_of_obc,
+        pvc_size=None,
+        delay=60,
+        run_time=1440,
+    ):
+        """
+        Concurrent bulk operations of following
+            PVC creation - all supported types  (RBD, CephFS, RBD-block)
+            PVC deletion - all supported types  (RBD, CephFS, RBD-block)
+            OBC creation
+            OBC deletion
+            APP pod creation - all supported types  (RBD, CephFS, RBD-block)
+            APP pod deletion - all supported types  (RBD, CephFS, RBD-block)
+
+        Args:
+            project_factory : Fixture to create a new Project.
+            num_of_pvc (int): Bulk PVC count
+            num_of_obc (int): Bulk OBC count
+            pvc_size (str): size of all pvcs to be created with Gi suffix (e.g. 10Gi).
+            If None, random size pvc will be created
+            delay (int): Delay in seconds before starting the next cycle
+            run_time (int): The amount of time the particular stage has to run (in minutes)
+
+        """
+        end_time = datetime.now() + timedelta(minutes=run_time)
+        cycle_count = 1
```

Comment:
I see the cycle count being incremental in line 778, but Can we get cycle_count as parameter in stage3() ?? Better to pass it via testcase and have pre defined cycles per stage?

Reply:
Handling the stage via cycle may not help in our case as we are aiming to run the stage for a longer duration of time. 
If we want to consider it based on the cycles we may have to first get how many such cycles would be required for running the identified duration and then pass the value. This would be a tedious task as there could be a  frequent requirement change.
In the current implementation, the number of cycles per stage depends on the run_time here.